packages/@cdklabs/cdk-cicd-wrapper/src/common/spi/ResourceContext.ts

```diff
@@ -61,9 +61,12 @@ class ScopedStorage {
    * @param provider The resource provider instance.
    */
   public add(name: string, provider: IResourceProvider): void {
-    if (this.providers.has(name)) {
+    if (this.resourceContext.locked && this.providers.has(name)) {
```

Comment:
For my own knowledge what does .locked signify and why only check if locked?

Reply:
why not check if just locked why also check if it has name?

Reply:
i understand what .locked signifies

Reply:
I don't want the providers to be modified once the stage generations started. It can cause strage results.

Reply:
so if it is locked but has no name you can modify it?

Reply:
@gyalai-aws ^